### PR TITLE
[interfaces]: Move IP/MTU information from interfaces file into database

### PR DESCRIPTION
**Pull request by stcheng**
- Move front panel ports and port channels MTU and IP configurations out of
the current /etc/network/interfaces file and store them in the configuration
database.

- The default MTU value for both front panel ports and the port channels is
9100. Add the option to indicate the desired MTU settings in the port_config.ini
file.

- Introduce portmgrd which will pick up the MTU configurations from the
configuration database.

- The updated intfmgrd will pick up IP address changes from the configuration
database.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

### CODE REVIEW

File: src/sonic-config-engine/portconfig.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def parse_port_config_file(port_config_file):
     ports = {}
     port_alias_map = {}
     # Default column definition
-    titles = ['name', 'lanes', 'alias', 'index']
+    titles = ['name', 'lanes', 'alias', 'index', 'mtu']
```
Comment (prsunny) [inline src/sonic-config-engine/portconfig.py]: As we discussed, we won't be needing mtu in portconfig.ini.

**Comment by lguohan** [comment]
change virtual switch as well.

can you specify the dependency of your commit?

File: files/image_config/interfaces/interfaces.j2 [hunk 2]
```diff
@@ -52,10 +52,9 @@ iface eth0 inet dhcp
 # The switch front panel interfaces
 {% for (name, prefix) in INTERFACE %}
 allow-hotplug {{ name }}
-iface {{ name }} {{ 'inet' if prefix | ipv4 else 'inet6' }} static
-    mtu {{ PORT[name]['mtu'] if PORT[name]['mtu'] else 9100 }}
-    address {{ prefix | ip }}
-    netmask {{ prefix | netmask if prefix | ipv4 else prefix | prefixlen }}
+iface {{ name }} {{ 'inet' if prefix | ipv4 else 'inet6' }} manual
```
Comment (lguohan) [inline files/image_config/interfaces/interfaces.j2]: why do we still keep up/down in the interfaces.j2? I thought we are going to remove all of them from the interfaces.j2?

**Comment by stcheng** [comment]
this pull request requires the following three commits from sonic-swss as well:
https://github.com/Azure/sonic-swss/pull/574
https://github.com/Azure/sonic-swss/pull/574

**Comment by lguohan** [comment]
@stcheng , all deps merged, can you update the submodule?

**Comment by stcheng** [comment]
@lguohan, updated

**Comment by nikos-github** [comment]
@stcheng @lguohan @zhenggen-xu @jipanyang With this change, existing configurations are breaking on upgrade because config_db.json didn't require the admin_status (just upgraded my setup and all the interfaces were down). Any reason why we can't make the default for the port to be up if config_db.json doesn't have the admin_status to be down?

**Comment by zhenggen-xu** [comment]
@nikos-github  I haven't tried the image with this checkin, in general, if the port is part of a VLAN or have IP address configured, it would be brought up during the initialization. I assume that was the case since you just did the upgrade.  But can you double check if orchagent etc is running fine?

**Comment by nikos-github** [comment]
@zhenggen-xu The specific configs I had were only for front panel ports. I upgraded to newer image and all interfaces were admin down. This change as it stands will break all devices when existing configs are deployed on upgrade.

**Comment by zhenggen-xu** [comment]
If ports have IPs but are down due to the change of moving things to DB, that is a regression bug needs to be fixed.  @stcheng can you confirm if that could be the case?

**Comment by stcheng** [comment]
@zhenggen-xu yes. i have set the default configuration of the admin status as up when generating the configuration from the minigraph. however, when inheriting the old configurations, it will cause the regression issue that the ports are not brought up. @nikos-github do you have any suggestions of fixing this?

**Pull request merged**